packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -492,7 +492,7 @@ export async function reconcileTransactions(
       // fields.
       fuzzyDataset = await db.all(
         `SELECT id, is_parent, date, imported_id, payee, category, notes, reconciled FROM v_transactions
-           WHERE date >= ? AND date <= ? AND amount = ? AND account = ?`,
+           WHERE date >= ? AND date <= ? AND amount = ? AND account = ? AND imported_id IS NULL`,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏this is an interesting proposal, but I think it would break an existing use-case.

There are folks that do both bank-sync + manual imports (don't ask me why.. I couldn't explain that. It's just my observation from the questions on discord). So for these folks the bank-sync imports would work fine with your patch. However, then the subsequent manual csv imports would break as they would create duplicates.

(also semi-related note - lately I've been considering enabling the "import" button even for accounts that have bank-sync enabled)

There is a new addition to the import modal to skip the reconciliation logic entirely. It's a checkbox you can turn on. Would using that solve your problem?

Reply:
I might be able to shed some light on the reasons:
1. not everyone will be interacting with actual every single month & several banks heavily restrict how far syncs go back, so there might be gaps with _just_ sync (me, sometimes :) )
2. some people might not trust the syncing (from either Actual or their bank's side) enough and want to manually import their year overview (f.e.)

Reply:
> There is a new addition to the import modal to skip the reconciliation logic entirely. It's a checkbox you can turn on. 

Disabling all reconciliation / dedup is too aggressive of a fix for #2562 .

In cases where people trust the data they're importing, and they trust that the imported IDs will correctly match, they just want to avoid magic heuristics from silently deleting their transactions.